src/OVAL/probes/unix/process58.c

```diff
@@ -351,6 +352,83 @@ static int get_exec_shield_status(int pid) {
 	return ret;
 }
 
+/**
+ * Parse /proc/%d/cmdline file
+ * @param filepath Path to file ~ use preallocated buffer for the path
+ * @param buffer output buffer with non-zero size
+ * @return ps-like command info or NULL
+ */
+static inline bool get_process_cmdline(const char* filepath, struct oscap_buffer* const buffer){
+
+	int fd = open(filepath, O_RDONLY, 0);
+
+	if (fd < 0) {
+		return false;
+	}
+
+	oscap_buffer_clear(buffer);
+
+
+
+	for(;;) {
+		static const int chunk_size = 1024;
+		char chunk[chunk_size];
+		// Read data, store to buffer
+		ssize_t read_size = read(fd, chunk, chunk_size );
```

Comment:
`(read_size > 0 && read_size <= chunk_size)` indicates EOF, but `-1` indicates error. If error happened while reading we should react accordingly and not continue as if nothing happened.